src/coreclr/jit/valuenum.cpp

```diff
@@ -4594,6 +4594,88 @@ bool ValueNumStore::IsVNConstant(ValueNum vn)
     }
 }
 
+bool ValueNumStore::IsVNVectorZero(ValueNum vn)
+{
+    if (vn == NoVN)
+    {
+        return false;
+    }
+#if FEATURE_HW_INTRINSICS
+    VNFuncApp funcApp;
+    if (GetVNFunc(vn, &funcApp) && funcApp.m_arity == 1)
+    {
+        switch (funcApp.m_func)
+        {
+            case VNF_HWI_Vector128_get_Zero:
+#if defined(TARGET_XARCH)
+            case VNF_HWI_Vector256_get_Zero:
+#elif defined(TARGET_ARM64)
+            case VNF_HWI_Vector64_get_Zero:
+#endif
+            {
+                return true;
+            }
+
+            default:
+                return false;
+        }
+    }
+#endif
+    return false;
+}
+
+#if FEATURE_SIMD
+VNSimdTypeInfo ValueNumStore::GetSimdTypeOfVN(ValueNum vn)
```

Comment:
Nit: needs a function header.

Reply:
There should be one.

Reply:
I think @SingleAccretion is referring to the function header comment: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#94-function-header-comment
It looks like it is missing on a lot of these functions.

Reply:
Ah, I see. I can add this style of header comment then; most of the comments on other functions just simply use `//` so I was following what was already there.